src/hotspot/share/gc/g1/g1Policy.cpp

```diff
@@ -601,7 +601,7 @@ bool G1Policy::need_to_start_conc_mark(const char* source, size_t alloc_word_siz
 
   bool result = false;
   if (marking_request_bytes > marking_initiating_used_threshold) {
-    result = collector_state()->in_young_only_phase() && !collector_state()->in_young_gc_before_mixed();
+    result = collector_state()->in_young_only_phase();
```

Comment:
To me this looks like a change in behavior. The check you have removed is there to prevent requesting a new concurrent cycle when in the "Prepare Mixed" collections (the young GC preceding mixed collections).

Reply:
I think when it comes here, collector_state()->in_young_gc_before_mixed() should be definite, as about_to_start_mixed_phase() is false (collector_state()->in_young_gc_before_mixed() is part of the condition about_to_start_mixed_phase()).
How do you think about it?

Reply:
Thanks for catching this, I have fixed it.